pkg/sdk/extract.go

```diff
@@ -147,13 +170,20 @@ func (e *extractRequestPool) Get(requestIndex int) ExtractRequest {
 	return r
 }
 
+func (e *extractRequestPool) SetOffsetArrayPtrs(startArrayPtr, lengthArrayPtr unsafe.Pointer) {
+	e.startArrayPtr = startArrayPtr
+	e.lengthArrayPtr = lengthArrayPtr
+}
+
 func (e *extractRequestPool) Free() {
 	for _, v := range e.reqs {
 		for _, b := range v.resStrBufs {
 			b.Free()
 		}
 		C.free(unsafe.Pointer(v.resBuf))
 	}
+	C.free(e.startArrayPtr)
+	C.free(e.lengthArrayPtr)
```

Comment:
This is precisely what I wanted to do from the beginning. Unfortunately, it's not enough since memory is allocated multiple times here by `calloc`  in [plugin_extract_fields_sync()](https://github.com/falcosecurity/plugin-sdk-go/pull/103/files#diff-ba30d1c433f8704ceae2513ca7bd9fda642887423be4b0d64cb3305b1e4d47eaR54-R55) (which is invoked on every extraction request) and freed only once here (when the `extractRequestPool` gets cleaned up by [plugin_close()](https://github.com/falcosecurity/plugin-sdk-go/blob/main/pkg/sdk/symbols/open/open.go#L96-L108)).